### PR TITLE
Defect 26 unfiltered file upload

### PR DESCRIPTION
**Pull request by LJBabbage**
### What is the context of this PR?
To make sure someone isn't trying to land exe files on ONS estate. The code makes use of the python-magic library

### How to review 
Run the test and then on cf try to upload an exe with an xls extension

### CODE REVIEW

File: swagger_server/controllers/survey_response.py [hunk 1]
```diff
@@ -35,7 +37,8 @@ def add_survey_response(self, case_id, file):
 
         if case_id and file and hasattr(file, 'filename'):
             file_name, file_extension = os.path.splitext(file.filename)
-            is_valid_file, msg = self._is_valid_file(file_name, file_extension)
+            file_contents = file.read()
```
Comment (jonmorphic) [inline swagger_server/controllers/survey_response.py]: Is there are a risk here of reading large files into memory? I think to extract the magic number from the file, we may be able to read the first few bytes.

**Comment by pricem14pc** [comment]
Why aren't we simply checking the mime type? Current RRM classic checks for application/vnd.openxmlformats-officedocument.spreadsheetml.sheet or application/vnd.ms-excel

How do we know that magic.from_buffer(file_contents) reliably resolves all xls / xlsx files (created with all client applications/platforms) to 'Composite Document File V2 Document'?

Does the message 'The file extension has been altered' refer to the respondent having  uploaded a different survey response file type to the collection instrument file type they downloaded? This may well be possible. They may have saved their survey response as an xls instead of xlsx. This would be a valid acceptable scenario. Or is this message misleading?

File: swagger_server/test/test_survey_response.py [hunk 2]
```diff
@@ -144,6 +145,18 @@ def test_add_survey_response_long_file_name(self):
         self.assertEquals(status, 400)
         self.assertEquals(FILE_NAME_LENGTH_ERROR, msg)
 
+    def test_add_survey_response_file_extension_changed(self):
```
Comment (jonmorphic) [inline swagger_server/test/test_survey_response.py]: 👍

File: swagger_server/controllers/survey_response.py (hunk 1 above)
Comment (jonmorphic) [inline swagger_server/controllers/survey_response.py]: We probably want to use secure_filename here also. See http://werkzeug.pocoo.org/docs/0.12/utils/

**Comment by jonmorphic** [comment]
I think the mime type gets set on the client system, the exact algorithm varies, but generally based off file extension. Checking the magic number server side is probably safer.

**Comment by LJBabbage** [comment]
The mime time can be easily manipulated, we need to check the file

File: swagger_server/controllers/survey_response.py (hunk 1 above)
Comment (LJBabbage) [reply swagger_server/controllers/survey_response.py]: Done

Comment (LJBabbage) [reply swagger_server/controllers/survey_response.py]: We have to use the full file later

**Comment by pricem14pc** [comment]
Approving but will be speaking to Dan O'Brien re: IA requirements.
Dependent on OS linux libmagic

**Pull request merged**